### PR TITLE
updating documentation for CRF

### LINKED ISSUE #337: Request for example: CRF
This is pending #22 and #314 but once it's merged we should be sure to add some examples as it's highly requested.

### PR DESCRIPTION
**Pull request by Harsh188**
# Description

Improving documentation for `layers/crf.py`

Fixes part of #337 

## Type of change

- [ ] Bug fix
- [ ] New Tutorial
- [x] Updated or additional documentation
- [ ] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [x] By running Black + Flake8
    - [x] By running pre-commit hooks
- [x] This PR addresses an already submitted issue for TensorFlow Addons
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

### CODE REVIEW

**Comment by Harsh188** [comment]
I have no idea what's wrong with this. When I run `pytest -v --doctest-modules tensorflow_addons/layers/crf.py` I get no errors.

File: tensorflow_addons/layers/crf.py [hunk 1]
```diff
@@ -27,8 +27,37 @@
 class CRF(tf.keras.layers.Layer):
     """Linear chain conditional random field (CRF).
 
+    Inherits from: `tf.keras.layers.Layer`
```
Comment (WindQAQ) [inline tensorflow_addons/layers/crf.py:30]: Inherits from: `tf.keras.layers.Layer`.

File: tensorflow_addons/layers/crf.py [hunk 2]
```diff
@@ -27,8 +27,37 @@
...
         - [Conditional Random Field](https://en.wikipedia.org/wiki/Conditional_random_field)
+
+    Example:
+
+    >>> output = CRF(12)
+    >>> output.get_config() == {'name': 'crf', 'trainable': True, 'dtype': 'float32',
+    ... 'units': 12, 'chain_initializer': {'class_name': 'Orthogonal',
+    ... 'config': {'gain': 1.0, 'seed': None}}, 'use_boundary': True,
+    ... 'boundary_initializer': {'class_name': 'Zeros', 'config': {}}, 'use_kernel': True}
+    True
```
Comment (WindQAQ) [inline tensorflow_addons/layers/crf.py:42]: The example is not so useful. Can we have
```
layer = CRF(4)
inputs = xxx
layer(inputs)
```
or something like this?

File: tensorflow_addons/layers/crf.py [hunk 3]
```diff
@@ -27,8 +27,37 @@
...
+
+    Arguments:
+        units: Positive integer, dimensionality of the reservoir.
+        chain_initializer: Orthogonal matrix
+            Default: `orthogonal`
```
Comment (WindQAQ) [inline tensorflow_addons/layers/crf.py:47]:
```
        chain_initializer: Orthogonal matrix. Default to `orthogonal`.
```

File: tensorflow_addons/layers/crf.py [hunk 4]
```diff
@@ -27,8 +27,37 @@
...
+        use_boundary: `Boolean`, whether the layer uses a boundary vector.
+            Default: `True`
```
Comment (WindQAQ) [inline tensorflow_addons/layers/crf.py:49]:
```
        use_boundary: `Boolean`, whether the layer uses a boundary vector. Default to `True`.
```

File: tensorflow_addons/layers/crf.py [hunk 5]
```diff
@@ -27,8 +27,37 @@
...
+        boundary_initializer: Tensors initialized to 0
+            Default: `zeros`
```
Comment (WindQAQ) [inline tensorflow_addons/layers/crf.py:51]:
```
        boundary_initializer: Tensors initialized to 0. Default to `zeros`.
```

File: tensorflow_addons/layers/crf.py [hunk 6]
```diff
@@ -27,8 +27,37 @@
...
+        use_kernel: `Boolean`, whether the layer uses a kernel weights.
+            Default: `True`
```
Comment (WindQAQ) [inline tensorflow_addons/layers/crf.py:53]:
```
        use_kernel: `Boolean`, whether the layer uses a kernel weights. Default to `True`.
```

File: tensorflow_addons/layers/crf.py [hunk 7]
```diff
@@ -27,8 +27,37 @@
...
+    Call Arguments:
+        inputs: Positive integer, dimensionality of the output space.
+        mask: Tensor(shape=(batch_size, sequence_length), dtype=bool) or None
+            Default: `None`
```
Comment (WindQAQ) [inline tensorflow_addons/layers/crf.py:57]:
```
        mask: A boolean `Tensor` of shape `[batch_size, sequence_length]` or `None`. Default to `None`.
```

File: tensorflow_addons/layers/crf.py [hunk 8]
```diff
@@ -27,8 +27,37 @@
...
+    Raises:
+        ValueError: If input mask doesn't have dim 2 or None
+        NotImplementedError: If left padding is provided
```
Comment (WindQAQ) [inline tensorflow_addons/layers/crf.py:60]:
```
    Raises:
        ValueError: If input mask doesn't have dim 2 or `None`.
        NotImplementedError: If left padding is provided.
```

**Review by WindQAQ — COMMENTED** [review]
Thanks for the contribution!

File: tensorflow_addons/layers/crf.py (hunk 2 above)
Comment (WindQAQ) [reply tensorflow_addons/layers/crf.py:42]: @Harsh188 Just a reminder: here is also a comment.

Comment (Harsh188) [reply tensorflow_addons/layers/crf.py:42]: @WindQAQ Yep! Currently working on this but I've been running into issues where the format of the output is creating doctest issues.

Comment (Harsh188) [reply tensorflow_addons/layers/crf.py]: @WindQAQ I've had no luck with the doctest error. I posted a [message](https://gitter.im/tensorflow/sig-addons) on Gitter describing my issue can you take a look at it when you have time? Thanks

Comment (WindQAQ) [reply tensorflow_addons/layers/crf.py]: @Harsh188 

How about

```python
"""
    Example:

    >>> layer = tfa.layers.CRF(units=4)
    >>> inputs = np.random.rand(2, 4, 8).astype(np.float32)
    >>> decoded_sequence, potentials, sequence_length, chain_kernel = layer(inputs)
    >>> decoded_sequence.shape
    TensorShape([2, 4])
    >>> potentials.shape
    TensorShape([2, 4, 4])
    >>> sequence_length
    <tf.Tensor: shape=(2,), dtype=int64, numpy=array([4, 4])>
    >>> chain_kernel.shape
    TensorShape([4, 4])
"""
```

**Comment by WindQAQ** [comment]
Hi, I find that we forget to import `CRF` in `__init__.py`. Can you also import it in this PR?

```python
from tensorflow_addons.layers.crf import CRF
```

https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/__init__.py

File: tensorflow_addons/layers/crf.py (hunk 2 above)
Comment (Harsh188) [reply tensorflow_addons/layers/crf.py]: That's a lot cleaner than what I have currently thanks :)

**Review by WindQAQ — APPROVED** [review]

**Pull request merged**